packs/rackspace/actions/get_vm_ips.py

```diff
@@ -0,0 +1,44 @@
+from lib.action import PyraxBaseAction
+from lib.formatters import to_server_dict
+
+__all__ = [
+    'GetVMIPsAction'
+]
+
+class GetVMIPsAction(PyraxBaseAction):
+    def run(self, count=None, region=None, metadata=None):
+        if region:
+            cs = self.pyrax.connect_to_cloudservers(region=region)
+        else:
+            cs = self.pyrax.cloudservers
+
+        servers = cs.list()
+
+        result = []
+        for server in servers:
+            item = to_server_dict(server=server)
+
+            if metadata:
+                include = self._metadata_intersection(server=item,
+                                                      metadata=metadata)
+
+                if not include:
+                    continue
+
+            result.append(item['public_ips'][1])
```

Comment:
IIRC, it takes a while for a public IP to be assigned to a server after the create request so to be on the safe side, we should probably do something along the lines of:

```python

public_ips = item.get('public_ips', [])

if not public_ips:
  continue
```

If we don't do that, it's possible that this line will throw if someone just booted a VM...